joomla/joomla-cms
# [4.0] FilelistField/FolderlistField conflicting filter attribute

Pull Request for Issue #14899 .

### Summary of Changes

This replaces conflicting `filter` attribute in `Joomla\CMS\Form\Field\FilelistField` and `Joomla\CMS\Form\Field\FolderlistField` with new `fileFilter` and `folderFilter` attributes, respectively.

### Testing Instructions

Code review is fine, I guess. Or add filelist/folderlist field to a form, set fileFilter/folderFilter attribute and check that files/folders are filtered correctly. E.g.:

```
<field
	name="image"
	type="filelist"
	fileFilter="\.png$"
	directory="images/banners"
/>
```

### Documentation Changes Required

Yes. B/C break: existing forms using affected fields need to be updated with new attributes. Fields extending these fields need to be updated.
